contracts/token/ERC20/ERC20Channel.sol

```diff
@@ -0,0 +1,202 @@
+     final value, if the challenge time passes and the channel does not receive
+     a cooperativeClose request it would be able to be closed by the sender
+     transfering the final value requested.
+ */
+contract ERC20Channel is Ownable {
+  using SafeMath for uint256;
+
+  // Add recover method for bytes32 using ECRecovery lib from OpenZeppelin
+  using ECRecovery for bytes32;
+
+  // The ERC20 token to be used
+  ERC20 public token;
+
+  // The amount of time that a receiver has to challenge the sender
+  uint256 public challengeTime;
+
+  // The receiver of the tokens
+  address public receiver;
+
+  // The closing balance requested by the sender
+  uint256 public closingBalance;
+
+  // The timestamp of when the channel can be closed by the sender
+  uint256 public closeTime;
+
+  // Event triggered when the channel close is requested
+  event closeRequested(uint256 closeTime, uint256 closingBalance);
+
+  // Channel closed
+  event channelClosed(uint256 closeTime, uint256 finalBalance);
+
+  /**
+   * @dev Constructor
+   * @param tokenAddress address, the address of the ERC20 token that will be used
+   * @param _receiver address, the address of the receiver
+   * @param _challengeTime uint256, the time that a channel has before ends
+    with an uncooperative close
+   */
+  function ERC20Channel(address tokenAddress, address _receiver, uint256 _challengeTime) public {
+    require(tokenAddress != address(0));
+    require(_receiver != address(0));
+    require(_challengeTime > 0);
+
+    token = ERC20(tokenAddress);
+    challengeTime = _challengeTime;
+    receiver = _receiver;
+  }
+
+  /*
+   *  External functions
+   */
+
+  /**
+   * @dev Request an uncooperativeClose, it can be called only by the
+    sender/owner. It will save the closing balance requested and start the
+    challenge time period where the receiver can ask for a cooperativeClose.
+   * @param balance uint256, the final balance of the receiver
+   */
+  function uncooperativeClose(uint256 balance) external onlyOwner {
+    // Check that the closing request doesn't exist
+    require(closeTime == 0);
+
+    // Check that the balance is less or equal to the token balance
+    require(balance <= token.balanceOf(address(this)));
+
+    // Mark channel as closed and create closing request
+    closeTime = now.add(challengeTime);
+    closingBalance = balance;
+    closeRequested(closeTime, closingBalance);
+  }
+
+  /**
+   * @dev Close a channel with the agreement of the sender and receiver
+   * @param balance uint256, the final balance transfered of the channel
+   * @param balanceMsgSig bytes, the signature of the sender.
+    The msg signed by the receiver is generateBalanceHash(balance)
+   * @param closingSig bytes, the signature of the receiver
+    The msg signed by the sender is generateKeccak256(balanceMsgSig)
+   */
+  function cooperativeClose(uint256 balance, bytes balanceMsgSig, bytes closingSig) external {
```

Comment:
I'm not sure if we already discussed this, but taking a look at an implementation for [ETH payment channels](https://github.com/machinomy/machinomy-contracts/blob/master/contracts/Unidirectional.sol), there is indeed a cooperative close function that can be invoked by the receiver directly, which removes the need for the closingSig in favour of checking that `msg.sender == receiver`. Should we add it here as well?